srml/grandpa/src/lib.rs

```diff
@@ -238,7 +238,7 @@ decl_module! {
 					}
 				}
 
-				if block_number == pending_change.scheduled_at + pending_change.delay {
+				if block_number >= pending_change.scheduled_at + pending_change.delay {
```

Comment:
This is essentially patching around races between `on_finalize` implementations. I'm not sure this is the right approach; we are aware of the larger architectural issue and plan to introduce some changes to make this a non-issue.